libsupport/include/katana/ArrowVisitor.h

```diff
@@ -204,40 +205,24 @@ struct ArrowDispatcher {
     auto type_id =
         GetArrowTypeID(std::get<index>(std::forward_as_tuple(args...)));
 
-    switch (type_id) {
-#define TYPE_CASE(EnumType)                                                    \
-  case arrow::Type::EnumType: {                                                \
+#define ENUM_TYPE_CASE(z, n, unused)                                           \
+  case static_cast<arrow::Type::type>(n): {                                    \
```

Comment:
looks like z is also unused?

Reply:
That's accurate, it's an internal thing for the PP_REPEAT call (so that they can nest arbitrarily). I'll add a comment